activerecord/lib/active_record/encryption/encrypted_attribute_type.rb

```diff
@@ -135,8 +135,8 @@ def serialize_with_current(value)
 
         def encrypt_as_text(value)
           with_context do
-            encryptor.encrypt(value, **encryption_options).tap do |encrypted|
-              if !cast_type.binary? && encrypted.encoding == Encoding::BINARY
+            encryptor.encrypt(value, **encryption_options).tap do |encrypted_value|
+              if encryptor.encrypted?(encrypted_value) && !cast_type.binary? && encrypted_value.encoding == Encoding::BINARY
```

Comment:
For the standard Encryptor, `encrypted?` will deserialize the message to test , so I don't think we'd want to to that.

The point of this check is that data from the new `ActiveRecord::Encryption::MessagePackMessageSerializer` can only be safely stored in binary columns. 

So maybe instead of testing the encoding of the encrypted data, we need a new method on the encryptor - `binary?` and then we can have `!cast_type.binary? && encryptor.binary?`.

Reply:
> we need a new method on the encryptor - binary?

I'm guessing the method needs to be on the serializer? The encryptor can then delegate to it.

Reply:
Addressed in [these changes](https://github.com/rails/rails/compare/ded1018d883307279ad719d1aa06c24aa49559f8..62a140160712d3cfdda80d840d482808da658b1d).

It would be nicer if the encryptors and serializers had a base class, but I'll avoid adding one here to make it easier to review.